### PR TITLE
pf run create --file, input paths should be file names relative to cwd.

### PR DESCRIPTION
**Pull request by crazygao**
# Description

pf run create --file
input path should be file name relative to cwd

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_pf/_run.py [hunk 1]
```diff
@@ -559,7 +565,7 @@ def create_run(create_func: Callable, args):
             if not param:
                 continue
             params_override.append({param_key: param})
-
+            print(f"Overriding {param_key} with {param}")
```
Comment (D-W-) [inline src/promptflow/promptflow/_cli/_pf/_run.py:568]: Remove this?

Comment (crazygao) [reply src/promptflow/promptflow/_cli/_pf/_run.py:568]: Yes, just removing...

File: src/promptflow/promptflow/_cli/_pf/_run.py [hunk 2]
```diff
@@ -546,11 +548,15 @@ def create_run(create_func: Callable, args):
         column_mapping = list_of_dict_to_dict(column_mapping)
 
     if file:
+        # for file name items `flow` and `data`, convert them to relative path to the file
+        # data also has a special format `azureml:`, which should not be converted
         for param_key, param in {
             "name": name,
-            "flow": flow,
+            "flow": os.path.relpath(Path(flow), Path(file).parent) if flow else None,
```
Comment (D-W-) [inline src/promptflow/promptflow/_cli/_pf/_run.py:554]: Why change here? When load run from YAML, the base path will be set to file's folder

File: src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py [hunk 3]
```diff
@@ -67,17 +67,24 @@ def test_pf_version(self, capfd):
...
-            "--flow",
-            f"{FLOWS_DIR}/web_classification",
+            "--file",
+            f"{NEW_RUNS_DIR}/sample_bulk_run.yaml",
             "--data",
-            f"{DATAS_DIR}/webClassification3.jsonl",
+            f"{NEW_DATA_DIR}/webClassification3.jsonl",
             "--name",
             str(uuid.uuid4()),
+            "--flow",
```
Comment (D-W-) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py:85]: Maybe we need to discuss about the behavior here. I think we are only replacing YAML contents in param overrides. So the path provided here should be relative to the YAML file

File: src/promptflow/promptflow/_cli/_pf/_run.py (hunk 2 above)
Comment (crazygao) [reply src/promptflow/promptflow/_cli/_pf/_run.py:555]: let me check, I suspect we could change this by paramter --flow

**Pull request closed without merging**